### DISCUSSION TITLE
How to use DTO class outside of route method?

### DISCUSSION THREAD

**Discussion by pavdwest** [litestar-org/litestar#3898] [Q&A] [answered]
How can I get the following example working?
The seed_data method is called e.g. during startup, or from some other internal process. The instantiation of the DTO requires an asgi_connection: A ASGIConnection. This requirement doesn't make sense to me having worked with DTO patterns in multiple other frameworks. How can I instantiate DTOs in code using just data, and no unnecessary dependencies?

```
from dataclasses import dataclass

from litestar.dto import DataclassDTO, DTOConfig


@dataclass
class Item:
    id: int
    name: str
    secret: str
    

class ItemCreateDTO(DataclassDTO[Item]):
    config = DTOConfig(exclude=["id"])
    
    
async def seed_data():
    item1 = ItemCreateDTO(name="Item 1", secret="Secret 1")
    ...
```

**Comment by provinzkraut** [comment, marked as answer]
> This requirement doesn't make sense to me having worked with DTO patterns in multiple other frameworks.

Maybe not, but it is how they are currently working. So to answer your question: You can't.

There are plans to make the DTOs available standalone, but it will be a while before we get to that.

**Comment by pavdwest** [comment]
Ah ok - thank you very much for the swift response @provinzkraut - looking forward to the future developments on DTOs.
